<!DOCTYPE html>
<html>
	<body>

		<h2>JavaScript Errors</h2>

		<h3>Async Await</h3>

		<p id="demo"></p>
		<p id="demo1"></p>

		<script>			
			function resolveAfter2Seconds(x) {
				return new Promise((resolve, reject) => {
					try {
						setTimeoutt(() => {//fix here: setTimeoutt -> setTimeout
							resolve(x*2);
						}, 1000);
					}catch {
						reject(TypeError("Ol치, deu erro"));
					}
				});
			}

			async function f1() {
				try {
					var x = await resolveAfter2Seconds(10);
					//o c칩digo aguarda a resposta do async await, para depois continuar o c칩digo
					document.getElementById("demo").innerText = x;	 // 10
					document.getElementById("demo1").innerText = "Ol치";
				}catch(err) {
					document.getElementById("demo").innerText = err.name;
					document.getElementById("demo1").innerText = err.message;
				}
			}
			f1();
		</script>

	</body>
</html>
